src/main/scala/io/iohk/ethereum/consensus/pow/validators/OmmersValidator.scala

```diff
@@ -30,11 +30,16 @@ trait OmmersValidator {
   )(implicit blockchainConfig: BlockchainConfig): Either[OmmersError, OmmersValid] = {
 
     val getBlockHeaderByHash: ByteString => Option[BlockHeader] = blockchainReader.getBlockHeaderByHash
-    val bestBranch = blockchainReader.getBestBranch()
     val getNBlocksBack: (ByteString, Int) => List[Block] =
       (_, n) =>
-        ((blockNumber - n) until blockNumber).toList
-          .flatMap(nb => bestBranch.getBlockByNumber(nb))
+        Iterator
+          .iterate(blockchainReader.getBlockByHash(parentHash))(
+            _.flatMap(block => blockchainReader.getBlockByHash(block.header.parentHash))
+          )
+          .take(n)
+          .collect { case Some(block) => block }
```

Comment:
Is the consistency of the chain checked elsewhere ?
In case of getBlockByHash() returning a None, we may end up with shorter than expected block list, but it doesn't seem to be checked (within the StdOmmerValidator at least).

Reply:
Currently, the assumption is that only the canonical chain is stored. So we should have a parent (otherwise it means that the app is in an inconsistent state)

Reply:
We already have a rudimentary consistency check in #1070. 

Indeed, for the first few blocks it would be expected to have less than `n` blocks. I will try to make that more explicit in the code.